returnn/datasets/distrib_files.py

```diff
@@ -245,8 +267,32 @@ def init_seq_order(self, epoch: Optional[int] = None, seq_list=None, seq_order=N
         self._lazy_init_file_sizes()
         self._lazy_init_file_cache_proc()
 
-        # Cleanup and fill _files_order_cache.
         full_epoch_0idx = (epoch - 1) // self.partition_epoch
+
+        # Shard files across GPU workers
+        for k in list(self._files_for_this_worker.keys()):
+            if k < full_epoch_0idx:
+                del self._files_for_this_worker[k]
+        for ep_ in range(epoch, epoch + self.preload_next_n_sub_epochs + 1):
+            full_epoch_0idx_ = (ep_ - 1) // self.partition_epoch
+            if full_epoch_0idx_ in self._files_for_this_worker:
+                continue
+            if self.seq_ordering == "default":
+                files = self.files
```

Comment:
This is wrong here. Sharding should always be done, independent of the ordering.

Reply:
But it is always done, see L290-293

Reply:
Ah right, sorry. But see my other comment on this logic.